apps/dc_tools/odc/apps/dc_tools/s3_to_dc.py

```diff
@@ -152,6 +152,12 @@ def cli(
 
     # Consume generator and fetch YAML's
     dc = Datacube()
+    odc_products = dc.list_products().name.values
+
+    assert set(candidate_products).issubset(
+        set(odc_products)
+    ), f"{candidate_products} is not a subset of {odc_products}"
+
```

Comment:
```suggestion
    odc_products = set(odc_products)
    if not set(candidate_products).issubset(odc_products):                                
        missing_products = list(set(candidate_products) - odc_products)                   
        print(f"Error: Requested Product/s {', '.join(missing_products)} {'is' if len(missing_products) == 1 else 'are'} "
               "not present in the ODC Database", file=sys.stderr)
        sys.exit(1)

```

Reply:
For some context, I read about assertions. And they should ONLY be used for conditions that are not EVER expected to be met! 

For things that we expect happen, exceptions are better.